pkgs/development/php-packages/psysh/default.nix

```diff
@@ -1,22 +1,52 @@
 {
   fetchFromGitHub,
+  fetchurl,
   lib,
   php,
 }:
 
-php.buildComposerProject (finalAttrs: {
+let
   pname = "psysh";
-  version = "0.11.21";
+  version = "0.12.3";
 
   src = fetchFromGitHub {
     owner = "bobthecow";
     repo = "psysh";
-    rev = "v${finalAttrs.version}";
-    hash = "sha256-YuBn4mrgOzGeMGfGcyZySAISmQdv3WRGn91PRozyxdI=";
+    rev = "v${version}";
+    hash = "sha256-v2UAhxnfnVwA05sxcqMU5vmQcwlBcc901PYJqYf+pCw=";
   };
 
-  composerLock = ./composer.lock;
-  vendorHash = "sha256-FZFeO7UiVssxTf0JX6wdjrAE+jucYnfQJA1eOng39lQ=";
+  composerLock = fetchurl {
+    name = "composer.lock";
+    url = "https://github.com/bobthecow/psysh/releases/download/v${version}/composer-v${version}.lock";
+    hash = "sha256-ur6mzla3uXeFL6aEHAPdpxGdvcgzOgTLW/CKPbNqeCg=";
+  };
+in
+php.buildComposerProject2 (finalAttrs: {
+  inherit
+    pname
+    version
+    composerLock
+    src
+    ;
```

Comment:
If we just inherit everything, it is a bit cleaner to just write it directly here and use finalAttrs to get the variables into mkComposerVendor

Reply:
Yes, I thought about that already and that's possible, but we don't inherit the phases. What do you suggest?